### PR TITLE
pillarnet

### PR DESCRIPTION
**Pull request by agent-sgs**
I have add the PillarNet componets: DynamicPillarPFE, spconv_backbone_2d, BaseBackboneV1. The training schedules for KITTI and Waymo are similar to SECOND and CenterPoint resprectively.

### CODE REVIEW

**Comment by jihanyang** [comment]
Thanks for your PR. The implementation is very clear. Could you also provide the training config and add pretrained model as well as model performance to the model zoo?

File: pcdet/models/backbones_3d/vfe/dynamic_pillar_vfe.py [hunk 1]
```diff
@@ -140,3 +140,101 @@ def forward(self, batch_dict, **kwargs):
         batch_dict['pillar_features'] = features
         batch_dict['voxel_coords'] = voxel_coords
         return batch_dict
+
+
+class DynamicPillarPFE(VFETemplate):
```
Comment (sshaoshuai) [inline pcdet/models/backbones_3d/vfe/dynamic_pillar_vfe.py:145]: What's the difference between this DynamicPillarPFE and  the original DynamicPillarVFE? 
Can we use the original DynamicPillarVFE and add some config to support the function of `DynamicPillarPFE`?

**Review by sshaoshuai — COMMENTED** [review]
Thanks very much for your nice PR. I also think adding the waymo configs of PillarNet would be better.

File: pcdet/models/backbones_3d/vfe/dynamic_pillar_vfe.py (hunk 1 above)
Comment (agent-sgs) [reply pcdet/models/backbones_3d/vfe/dynamic_pillar_vfe.py:145]: DynamicPillarPFE  is the simplied version for dynamic pillar representation, where f_cluster is also dropped and voxel_coords is 3-dimensions (byx) for spconv 2D.
Also, renaming voxel coords as pillar coords may be better.

**Comment by OpenMMLab-Assistant-007** [comment]
Hi @agent-sgs   !
First of all, we want to express our gratitude for your significant PR in the OpenMMLab project. Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR.

We would also like to invite you to join our Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/UjgXkPWNqA

If you have WeChat account，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤

**Pull request merged**